airbyte-integrations/connectors/source-freshdesk/source_freshdesk/source.py

```diff
@@ -52,53 +51,45 @@ def __init__(self, api_key: str) -> None:
 
 
 class SourceFreshdesk(AbstractSource):
-    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
-        alive = True
-        error_msg = None
+    @staticmethod
+    def _get_stream_kwargs(config: Mapping[str, Any]) -> dict:
+        return {"authenticator": FreshdeskAuth(config["api_key"]), "config": config}
+
+    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]):
         try:
-            url = urljoin(f"https://{config['domain'].rstrip('/')}", "/api/v2/settings/helpdesk")
-            response = requests.get(url=url, auth=FreshdeskAuth(config["api_key"]))
-            response.raise_for_status()
-        except requests.HTTPError as error:
-            alive = False
-            body = error.response.json()
-            error_msg = f"{body.get('code')}: {body.get('message')}"
+            check_stream = CheckStream(stream_names=["settings"], options={})
+            return check_stream.check_connection(self, logger, config)
         except Exception as error:
-            alive = False
-            error_msg = repr(error)
-
-        return alive, error_msg
+            return False, repr(error)
```

Comment:
This (and the `try:`) is not necessary - [CheckStream will handle this part](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py#L42-L50)

Reply:
We have a `test_check_connection_empty_config` unittest that runs `check_connection` method on a `Source`. In this test we pass a wrong config value (config is missing `api_key`) to the Source and we are failing without try/except block

Reply:
It happen right before try/except in cdk method. This is failing row:
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py#L32

Reply:
Got it - looks like something CheckStream should handle, I can look into it in the CDK

Reply:
Looks like this is specific to the source because we need the `config` to create the authenticator in the `streams` method. So I think it makes sense to keep the try/excepting the way that it is 👍🏻